spec/components/govuk_component/header_component_spec.rb

```diff
@@ -3,7 +3,7 @@
 RSpec.describe(GovukComponent::HeaderComponent, type: :component) do
   before do
     allow_any_instance_of(GovukComponent::HeaderComponent::NavigationItem).to(
-      receive(:request).and_return(double(ActionDispatch::Request, get?: true, path: current_page))
+      receive(:request).and_return(double(ActionDispatch::Request, get?: true, format: double(html?: true), path: current_page))
```

Comment:
Thanks @benilovj. Let me have a quick think about the approach for overriding the others, but maybe something like this makes it a bit more obvious what's going on 🤔 

```suggestion
      receive(:current_page?).and_wrap_original do |_original_method, path|
        path == current_page
      end
```